scheduler/src/cook/mesos/compute_cluster.clj

```diff
@@ -0,0 +1,105 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns cook.mesos.compute-cluster
+  (:require         [clojure.tools.logging :as log]
+                    [datomic.api :as d :refer (q)]))
+
+(defn- create-missing-compute-cluster
+  "Create a missing compute-cluster for one thats not yet in the database."
+  [conn compute-cluster]
+  (log/info "Installing a new compute cluster in datomic for " compute-cluster)
+  (d/transact
```

Comment:
`d/transact` returns a future...should we wait on that here? I'm just worried about repeated calls to `get-mesos-cluster-map` potentially creating multiple clusters.

Reply:
Yes, we should wait. This is going to be done sequentially during startup only, so we can't create clusters. But we may fail to startup and get random NPE's elsewhere if we lose this race, so good catch.